File: src/MistwoodTales.Game/Player.cpp
#include "Player.h"



Player::Player()
	: HP(0)
	, MaxHP(0)
{
	X = 120;
	Y = 20;
	Symbol = '%';
	FaceDirection = Right;
}


Player::~Player()
{
}


void Player::TakeDamage(int damage)
{
	HP -= damage;
	if (HP < 0) {
		HP = 0;
	}
}

File: src/MistwoodTales.Game.Client/Legacy/Player.cs
namespace MistwoodTales.Game.Client.Legacy
{
    public class Player: MoveableEntity
    {
        public Player()
        {
            Direction = Direction.Right;
            Symbol = '%';
            Point = new Point(120, 20);
        }

        public int HP { get; set; }
        public int MaxHP { get; set; }

        public Direction Direction { get; set; }
        public string Name { get; set; }

        public void TakeDamage(int damage)
        {
            HP -= damage;
            if (HP < 0)
            {
                HP = 0;
            }
        }
    }
}
